src/core/api/coap_api.cpp

```diff
@@ -148,13 +148,23 @@ const otCoapOption *otCoapHeaderGetNextOption(otCoapHeader *aHeader)
     return static_cast<const otCoapOption *>(static_cast<Coap::Header *>(aHeader)->GetNextOption());
 }
 
-otMessage *otCoapNewMessage(otInstance *aInstance, const otCoapHeader *aHeader)
+otMessage *otCoapNewMessage(otInstance *aInstance, const otCoapHeader *aHeader, otMessageSettings *aSettings)
 {
     Message * message;
     Instance &instance = *static_cast<Instance *>(aInstance);
+    uint8_t   priority;
 
     VerifyOrExit(aHeader != NULL, message = NULL);
-    message = instance.GetApplicationCoap().NewMessage(*(static_cast<const Coap::Header *>(aHeader)));
+
+    if (aSettings != NULL)
+    {
+        VerifyOrExit(aSettings->mLinkSecurityEnabled, message = NULL);
+        VerifyOrExit(aSettings->mPriority <= OT_MESSAGE_PRIORITY_HIGH, message = NULL);
+    }
+
+    priority = (aSettings == NULL) ? OT_MESSAGE_PRIORITY_NORMAL : aSettings->mPriority;
```

Comment:
Combine with above `if-statement` in `else` block?

Reply:
Will update it.